StyleCop.Analyzers/StyleCop.Analyzers/Lightup/SemanticModelExtensions.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Lightup
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using System.Threading;
+    using Microsoft.CodeAnalysis;
+
+    internal static class SemanticModelExtensions
+    {
+        private static readonly Func<SemanticModel, int, CancellationToken, ImmutableArray<IImportScopeWrapper>> GetImportScopesAccessor;
+
+        static SemanticModelExtensions()
+        {
+            CreateGetImportScopesAccessor(out var getImportScopesAccessor, out var supportsGetImportScopes);
+            GetImportScopesAccessor = getImportScopesAccessor;
+            SupportsGetImportScopes = supportsGetImportScopes;
+        }
+
+        public static bool SupportsGetImportScopes { get; }
+
+        public static ImmutableArray<IImportScopeWrapper> GetImportScopes(this SemanticModel semanticModel, int position, CancellationToken cancellationToken = default)
```

Comment:
I didn't add any lightup tests for this class, but relied on the SA1121 and SA1404 tests instead. Does that feel good enough?